### ISSUE TITLE
 Quantization aware training with full integer quantization support in tf.keras API

### ISSUE THREAD

**Issue by anilsathyan7** [tensorflow/model-optimization#600]
**System information**

- TensorFlow version (you are using): tf-nightly, 2.3.0
- Are you willing to contribute it (Yes/No): No

The quantization aware training [example](https://www.tensorflow.org/model_optimization/guide/quantization/training_example) shows us how to perform the quantization in tf,.keras, without affecting the accuracy of the model. However the trained model still accepts float inputs and outputs and has addtional **quantize and dequantize layers** to handle the inputs and outputs. Hardware accelerators/API's (NNAPI, Hexagon etc) requires you to have a **full integer quantized** models and currently it requires post-training quantization (representative dataset) or legacy quantization-aware training(TF V1). Also, some of these API's does not seem to support this addtional **Quantize** layers.

So, is this feature currently supported by tf.keras QAT API? If not, can we combine QAT with post training quantization to acheive similar results(i.e no accuracy loss).

If it's supprted we can convert the float models to full integer quantized format using the TF2, without considerable loss in accuracy and it would help developers to deploy the models using harrware accelartors like DSP, NPU etc.

Ref: https://github.com/tensorflow/tensorflow/issues/38285#issuecomment-640315447

**Comment by sunzhe09** [comment]
I also need the feature ！！

**Comment by anilsathyan7** [comment]
It looks like the feature is supported on tf 2.4 rc1 or above if you set **converter._experimental_new_quantizer = True**, during tflite conversion.
Refer: https://github.com/tensorflow/tensorflow/issues/42082#issuecomment-737074024
